cmd/snap/cmd_run_test.go

```diff
@@ -1583,3 +1716,106 @@ func (s *RunSuite) TestRunGdbserverNoGdbserver(c *check.C) {
 	_, err := snaprun.Parser(snaprun.Client()).ParseArgs([]string{"run", "--gdbserver", "snapname.app"})
 	c.Assert(err, check.ErrorMatches, "please install gdbserver on your system")
 }
+
+func (s *RunSuite) TestWaitInhibitUnlock(c *check.C) {
+	var called int
+	restore := snaprun.MockIsLocked(func(snapName string) (runinhibit.Hint, error) {
+		called++
+		if called < 5 {
+			return runinhibit.HintInhibitedForRefresh, nil
+		}
+		return runinhibit.HintNotInhibited, nil
+	})
+	defer restore()
+
+	notInhibited, err := snaprun.WaitInhibitUnlock("some-snap", runinhibit.HintNotInhibited, nil)
+	c.Assert(err, check.IsNil)
+	c.Check(notInhibited, check.Equals, true)
+	c.Check(called, check.Equals, 5)
+}
+
+func (s *RunSuite) TestWaitInhibitUnlockWaitsForSpecificHint(c *check.C) {
+	var called int
+	restore := snaprun.MockIsLocked(func(snapName string) (runinhibit.Hint, error) {
+		called++
+		if called < 5 {
+			return runinhibit.HintInhibitedGateRefresh, nil
+		}
+		return runinhibit.HintInhibitedForRefresh, nil
+	})
+	defer restore()
+
+	notInhibited, err := snaprun.WaitInhibitUnlock("some-snap", runinhibit.HintInhibitedForRefresh, nil)
+	c.Assert(err, check.IsNil)
+	c.Check(notInhibited, check.Equals, false)
+	c.Check(called, check.Equals, 5)
+}
+
+func (s *RunSuite) TestWaitInhibitUnlockWithErrorChannel(c *check.C) {
+	errCh := make(chan error, 1)
+	var called int
+	restore := snaprun.MockIsLocked(func(snapName string) (runinhibit.Hint, error) {
+		called++
+		if called == 1 {
+			errCh <- fmt.Errorf("boom")
+		}
+		return runinhibit.HintInhibitedForRefresh, nil
+	})
+	defer restore()
+
+	notInhibited, err := snaprun.WaitInhibitUnlock("some-snap", runinhibit.HintNotInhibited, errCh)
+	c.Assert(err, check.IsNil)
+	c.Check(notInhibited, check.Equals, false)
```

Comment:
this bit of the unit test confuses me, shouldn't if we have an error returned on the errCh also then have an error returned from WaitInhibitUnlock too? Otherwise snap run will just output `boom` on stderr and then try to run the snap anyways no?

Reply:
Yes, but that was the idea , this relates to e https://github.com/snapcore/snapd/pull/7700/files#r659738457

Reply:
ah okay sorry I missed that, this makes sense in that case. This is one of the pitfalls of having PR's open for so long haha